R/stat-sf.R

```diff
@@ -3,12 +3,52 @@
 #' @usage NULL
 #' @format NULL
 StatSf <- ggproto("StatSf", Stat,
-  compute_group = function(data, scales) {
-    bbox <- sf::st_bbox(data[[ geom_column(data) ]])
-    data$xmin <- bbox[["xmin"]]
-    data$xmax <- bbox[["xmax"]]
-    data$ymin <- bbox[["ymin"]]
-    data$ymax <- bbox[["ymax"]]
+  compute_layer = function(self, data, params, layout) {
+    # add coord to the params, so it can be forwarded to compute_group()
+    params$coord <- layout$coord
+    ggproto_parent(Stat, self)$compute_layer(data, params, layout)
+  },
+
+  compute_group = function(data, scales, coord) {
+    geometry_data <- data[[ geom_column(data) ]]
+    geometry_crs <- sf::st_crs(geometry_data)
+
+    bbox <- sf::st_bbox(geometry_data)
+
+    if (inherits(coord, "CoordSf")) {
+      # if the coord derives from CoordSf, then it
+      # needs to know about bounding boxes of geometry data
+      coord$record_bbox(
+        xmin = bbox[["xmin"]], xmax = bbox[["xmax"]],
+        ymin = bbox[["ymin"]], ymax = bbox[["ymax"]]
+      )
+
+      # to represent the location of the geometry in default coordinates,
+      # we take the mid-point along each side of the bounding box and
+      # backtransform
+      bbox_trans <- sf_transform_xy(
+        list(
+          x = c(rep(0.5*(bbox[["xmin"]] + bbox[["xmax"]]), 2), bbox[["xmin"]], bbox[["xmax"]]),
```

Comment:
Maybe it would be easier to read if we use `$` instead of `[[`?

Reply:
if `bbox` comes from `sf::st_bbox`, it is a numeric vector, not a list.

Reply:
Oh, thanks. But this is what I'm seeing...:

``` r
library(sf)
#> Linking to GEOS 3.8.0, GDAL 3.0.4, PROJ 6.2.1

a <- st_sf(a = 1:2, geom = st_sfc(st_point(0:1), st_point(1:2)), crs = 4326)
bbox <- st_bbox(a)
is(bbox)
#> [1] "bbox"
bbox$xmin
#> xmin 
#>    0
```

<sup>Created on 2020-02-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
You are right - a case of over-engineering from my side...

Reply:
So is this official API I can rely on? Should I change the code from `[[` to `$` as suggested?

Reply:
Sorry, though this seems convenient, now I don't feel it's very readable to use `$` for subsetting an element from a numeric vector. Please ignore, I just mistook `bbox` for a list, guessing from `[[`.